cfgov/prepaid_agreements/views.py

```diff
@@ -126,10 +138,36 @@ def index(request):
         'active_filters': available_filters,
         'valid_filters': valid_filters,
         'search_field': search_field,
+        'disclaimer_text': disclaimer_text,
+        'support_text': support_text
     })
 
 
+def get_detail_page_breadcrumb(request):
+    """
+    Determines link back to search page from detail page.
+    If referrer is search page and contains a query
+    string, returns referrer so query is preserved.
+    Otherwise, returns base search page path.
+    """
+    http_referer = request.META.get('HTTP_REFERER', '')
+    referrer = urlparse(http_referer)
+    search_page_path = reverse('prepaid_agreements:index')
+    if referrer.query and referrer.path == search_page_path:
+        return http_referer
+    else:
+        return search_page_path
+
+
 def detail(request, product_id):
+    disclaimer_text = ReusableText.objects.filter(
```

Comment:
Since these are defined above (in `validate_page_number`), could you make them variables at the top of the file?